src/classes/frDonation.cls

```diff
@@ -78,8 +79,25 @@ public class frDonation extends frModel {
         if(String.isBlank(o.StageName)) {
             o.StageName = (status == 'Failed') ? 'Closed Lost' : 'Closed Won';    
         }
+        
+        if(hasNonProfitSuccessPackage()){
+        	List<Contact> contacts = Database.query('select AccountId from Contact where fr_ID__c = :donorId');
+        	if(contacts.size() > 0){
+        		Contact primaryContact = contacts.get(0);
```

Comment:
This all looks good.  Can you add a check to only do this if the fields are not populated?  In the case where someone is already populating these fields on their own (somehow) I'd like to make sure we don't overwrite their changes. 

And also fix the indentation on line 87